chain/ethereum/ethererum_chain.go

```diff
@@ -345,6 +346,38 @@ func (c *EthereumChain) SendFunds(ctx context.Context, keyName string, amount ib
 	return err
 }
 
+type TransactionReceipt struct {
+	TxHash string `json:"transactionHash"`
+}
+
+func (c *EthereumChain) SendFundsWithNote(ctx context.Context, keyName string, amount ibc.WalletAmount, note string) (string, error) {
+	cmd := []string{"cast", "send", amount.Address, hexutil.Encode([]byte(note)), "--value", amount.Amount.String(), "--json"}
+	if keyName == "faucet" {
+		cmd = append(cmd,
+			"--private-key", "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
+			"--rpc-url", c.GetRPCAddress(),
+		)
+
+	} else {
+		cmd = append(cmd,
+			"--keystore", c.keystoreMap[keyName],
```

Comment:
UX nit: keyName in c.keystoreMap check?

Reply:
Here's the current error that would be returned.
exit code 1:  Error: Keystore file `""` does not exist

But I can add that check in the various areas to make it a little clearer we don't know that keyname since it only resolves to an empty string for the keystore file.